legacy/src/main/java/org/opensearch/sql/legacy/executor/join/ElasticJoinExecutor.java

```diff
@@ -99,6 +99,7 @@ public void run() throws IOException, SqlParseException {
       this.metaResults.setTookImMilli(joinTimeInMilli);
     } catch (Exception e) {
       LOG.error("Failed during join query run.", e);
+      throw new IllegalStateException("Error occurred during join query run", e);
```

Comment:
Any concern to append  `e.getMessage` to the error message? Ignore if this is the standard way in legacy code.

Reply:
Yes I see this as commonly followed practice to throw exceptions in legacy code and also most parts of the overall code where exception as a whole is used instead of exception message.